### ISSUE TITLE
why use negtive as positive in triplet loss

### ISSUE THREAD

**Issue by WilliamZhaoz** [clovaai/voxceleb_trainer#113]
Hi, thanks for your great work, I have a question
In triplet.py, why do you treat all negtive as positve and hard negtive as negtive ?
(
```
out_anchor = F.normalize(x[:, 0, :], p=2, dim=1)
out_positive = F.normalize(x[:, 1, :], p=2, dim=1) 
...
out_negative = out_positive[negidx, :] 
```
)

is it right? why it can work?

thanks
